packages/app/components/stack-modal/StackOrdersTable.tsx

```diff
@@ -87,62 +87,64 @@ const TableCowBody = ({
+          stackOrder.sellToken.decimals
+        );
+        const averagePrice = executedSellAmount / executedBuyAmount;
+
+        return (
+          <TableRow key={cowOrder.uid}>
+            <TableCell className="py-2 md:table-cell">
+              <BodyText
+                size={1}
+                className="text-primary-700 hover:underline hover:underline-offset-2"
               >
-                {addressShortner(cowOrder.uid)}
-              </Link>
-            </BodyText>
-          </TableCell>
-          <TableCell className="py-2">
-            <BodyText className="text-em-med" size={1}>
-              {formatDate(cowOrder.creationDate)}
-            </BodyText>
-          </TableCell>
-          <TableCell className="py-2 text-right ">
-            <BodyText className="text-em-med" size={1}>
-              {convertedAmount(
-                cowOrder.executedSellAmount,
-                stackOrder.sellToken.decimals
-              ).toFixed(4)}
-            </BodyText>
-          </TableCell>
-          <TableCell className="py-2 text-right ">
-            <BodyText className="text-em-med" size={1}>
-              {convertedAmount(
-                cowOrder.executedBuyAmount,
-                stackOrder.buyToken.decimals
-              ).toFixed(4)}
-            </BodyText>
-          </TableCell>
-          <TableCell className="hidden py-2 text-right md:table-cell">
-            {cowOrder.status === "fulfilled" ? (
+                <Link
+                  target="_blank"
+                  href={cowExplorerUrl(chainId, cowOrder.uid)}
+                >
+                  {addressShortner(cowOrder.uid)}
+                </Link>
+              </BodyText>
+            </TableCell>
+            <TableCell className="py-2">
+              <BodyText className="text-em-med" size={1}>
+                {formatDate(cowOrder.creationDate)}
+              </BodyText>
+            </TableCell>
+            <TableCell className="py-2 text-right ">
               <BodyText className="text-em-med" size={1}>
-                {averagePrice(cowOrder)}
+                {executedSellAmount.toFixed(4)}
               </BodyText>
-            ) : (
-              <BodyText className="text-gray-400 animate-pulse" size={1}>
-                fulfilling
+            </TableCell>
+            <TableCell className="py-2 text-right ">
+              <BodyText className="text-em-med" size={1}>
+                {executedBuyAmount.toFixed(4)}
               </BodyText>
-            )}
-          </TableCell>
-        </TableRow>
-      ))}
+            </TableCell>
+            <TableCell className="hidden py-2 text-right md:table-cell">
+              {cowOrder.status === "fulfilled" ? (
+                <BodyText className="text-em-med" size={1}>
+                  {averagePrice.toFixed(4)}
+                </BodyText>
+              ) : (
+                <BodyText className="text-gray-400 animate-pulse" size={1}>
+                  fulfilling
```

Comment:
Let's rename this to "Completing" maybe?

Also, we should check for "expired" status. Related to this issue https://linear.app/swaprhq/issue/STK-218/ui-should-show-if-it-is-a-cow-order-has-expired

Reply:
Will do, Maybe renaming `fulfilling` -> `in progress` feels better to me.